adal/tests/Test.ADAL.NET.Integration/DeviceProfileTests.cs

```diff
@@ -186,13 +186,17 @@ public void NegativeDeviceCodeTest()
         }
 
         [TestMethod]
-        public async Task NegativeDeviceCodeTimeoutTest()
+        public async Task NegativeDeviceCodeTimeoutTest_WithOneRetry()
         {
             MockHttpMessageHandler mockMessageHandler = new MockHttpMessageHandler(TestConstants.DefaultAuthorityHomeTenant)
             {
                 Method = HttpMethod.Get,
                 Url = TestConstants.DefaultAuthorityHomeTenant + "oauth2/devicecode",
-                ResponseMessage = MockHelpers.CreateSuccessDeviceCodeResponseMessage("1")
+                ResponseMessage = MockHelpers.CreateSuccessDeviceCodeResponseMessage(
+                    // do not lower this to 1-2s as test execution may be slow and the flow 
+                    // will never call the server
+                    expirationTimeInSeconds: 3, 
```

Comment:
If we use a much larger value here, say, `expirationTimeInSeconds: 30`, this test case would still pass, wouldn't it? If so, I would suggest to use a larger value here. Because, if value `1` or `2` would introduce those intermittent test failure, then `3` seems still too close. The purpose of this PR is to eleminate those failures, so why not do it in a more reliable way?